[en] Update images.md to follow style guide recommendations (#50774)
kubernetes/website: content/en/docs/concepts/containers/images.md

wait until at least one ongoing image pull is complete.

Limiting the number of parallel image pulls prevents image pulling from consuming
too much network bandwidth or disk I/O, when parallel image pulling is enabled.

You can set `maxParallelImagePulls` to a positive number that is greater than or
equal to 1. If you set `maxParallelImagePulls` to be greater than or equal to 2,
you must set `serializeImagePulls` to false. The kubelet will fail to start
with an invalid `maxParallelImagePulls` setting.

## Multi-architecture images with image indexes

As well as providing binary images, a container registry can also serve a
[container image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md).
An image index can point to multiple [image manifests](https://github.com/opencontainers/image-spec/blob/master/manifest.md)
for architecture-specific versions of a container. The idea is that you can have
a name for an image (for example: `pause`, `example/mycontainer`, `kube-apiserver`)
and allow different systems to fetch the right binary image for the machine
architecture they are using.

> Kubernetes itself typically names container images with a suffix `-$(ARCH)`. For

Reviewer: nit: could clarify that this is about Kubernetes releases and not the behavior of Kubernetes APIs / code.

backward compatibility, generate older images with suffixes. For instance,
an image named as `pause` would be a multi-architecture image containing manifests
for all supported architectures, while `pause-amd64` would be a backward-compatible
version for older configurations, or for YAML files with hardcoded image names
containing suffixes.

## Using a private registry

> Private registries may require keys to read images from them.

Reviewer: ```suggestion
Private registries may require authentication to be able to discover images and / or pull images from them.
```
?

Credentials can be provided in several ways:

- [Configuring Nodes to Authenticate to a Private Registry](#configuring-nodes-to-authenticate-to-a-private-registry)
  - all pods can read any configured private registries
  - requires node configuration by cluster administrator

> - [Kubelet Credential Provider to dynamically fetch credentials for private registries](#kubelet-credential-provider)

Reviewer: I suggest
```suggestion
- using a _kubelet credential provider_ plugin to [dynamically fetch credentials for private registries](#kubelet-credential-provider)
```

>   - kubelet can be configured to use credential provider exec plugin

Reviewer 2: ```suggestion
  
  kubelet can be configured to use credential provider exec plugin.
  
```
If only one item included, the bullet may be optional.

Author: Hi there, @windsonsea! 👋🏻 

Thanks for taking a look at this! 🙇🏻‍♀️ 

I changed this locally and noticed that removing the bullet causes the subitem to render side-by-side with the main item, like this:
![image](https://github.com/user-attachments/assets/1af6dd54-5bbc-440c-899c-1fb2a9718986)

Is that the result you wished?
Just wanted to check before I push the changes.

Thanks in advance!

Reviewer 2: Add blank lines around the `kubelet can be configured ...`

Author: Applied this change

    for the respective private registry.
- [Pre-pulled Images](#pre-pulled-images)
  - all pods can use any images cached on a node
  - requires root access to all nodes to set up

> - [Specifying ImagePullSecrets on a Pod](#specifying-imagepullsecrets-on-a-pod)

Reviewer: ```suggestion
- [Specifying imagePullSecrets when you define a Pod](#specifying-imagepullsecrets-on-a-pod)
```
?

Reviewer: The thing you type into a Pod manifest starts with lower case i.

>   - only pods which provide their own keys can access the private registry

Reviewer 2: ```suggestion
  
  Only pods which provide their own keys can access the private registry.
  
```

Author: Applied this change

### Kubelet credential provider for authenticated image pulls {#kubelet-credential-provider}

> {{< note >}}

Reviewer: I'd drop the note, and write a new paragraph about how this technique is useful for running static Pods (some of the other options may not work for static Pod launches)

}
```

Image pull operations pass the credentials to the CRI container runtime for every
valid pattern. For example, the following container image names would match
successfully:

> - `my-registry.io/images`

Reviewer: (Optional extra fix)
Use `example`, not `io` here.

{{< note >}}
{{< feature-state feature_gate_name="KubeletEnsureSecretPulledImages" >}}
Access to pre-pulled images may be authorized according to [image pull credential verification](#ensureimagepullcredentialverification).
{{< /note >}}

> 

Reviewer: Could mention applicability to static Pods here.

> ### Specifying `imagePullSecrets` on a Pod

Reviewer: [optional]
Can we move this earlier than the other mechanisms, given it's what we recommend?

Author: I moved this to the top. Applied the same change to the unordered list that links to it.

{{< note >}}
This is the recommended approach to run containers based on images
in private registries.
{{< /note >}}

Kubernetes supports specifying container image registry keys on a Pod.

> All `imagePullSecrets` must be Secrets that exist in the same namespace as the

Reviewer: Optional extra: use a glossary tool tip for "namespace".

Pod. These Secrets must be of type `kubernetes.io/dockercfg` or `kubernetes.io/dockerconfigjson`.

> #### Ensure Image Pull Credential Verification {#ensureimagepullcredentialverification}

Reviewer: Optional extra: switch this to sentence case

{{< feature-state feature_gate_name="KubeletEnsureSecretPulledImages" >}}

> If the `KubeletEnsureSecretPulledImages` feature gate is enabled, Kubernetes will

Reviewer: I think this is correct: 
```suggestion
If the `KubeletEnsureSecretPulledImages` feature gate is enabled for your cluster,
the kubelet instructs the container runtime to
```

1. Cluster with proprietary images, a few of which require stricter access control.
   - Ensure [AlwaysPullImages admission controller](/docs/reference/access-authn-authz/admission-controllers/#alwayspullimages)
     is active. Otherwise, all Pods potentially have access to all images.
   - Move sensitive data into a Secret resource, instead of packaging it in an image.
1. A multi-tenant cluster where each tenant needs own private registry.
   - Ensure [AlwaysPullImages admission controller](/docs/reference/access-authn-authz/admission-controllers/#alwayspullimages)
     is active. Otherwise, all Pods of all tenants potentially have access to all images.
   - Run a private registry with authorization required.
   - Generate registry credentials for each tenant, store into a Secret, and propagate
     the Secret to every tenant namespace.
   - The tenant then adds that Secret to `imagePullSecrets` of each namespace.

If you need access to multiple registries, you can create one Secret per registry.

> ## Legacy built-in kubelet credential provider

Reviewer: BTW we can't (shouldn't) direct people to read the docs aboutb this any more, as those docs are no longer served.

The rest of the text is good to keep.
